### ISSUE TITLE
Pretrained models for other backbone models

### ISSUE THREAD

**Issue by ChengshuLi** [fizyr/keras-retinanet#300]
Hi, 

Thank you for the great work!
Is there any chance you may release the pretrained models for other backbone models, e.g. resnet101, resnet152 or mobilenet128_1.0, mobilenet128_0.75, mobilenet160_1.0? Currently we only have pretrained models for resnet50. 

That would be super helpful for transfer learning. Otherwise, I might need to train on COCO from scratch. 

Thanks a lot!

**Comment by hgaiser** [comment]
Considering our resources for this project are limited, we don't provide the pretrained models for the other architectures. If in the future we happen to have trained these architectures on COCO we Will probably make them publicly available. For now, your best bet is to start with imagenet trained weights and then fine-tune on COCO or your own dataset.

**Comment by ChengshuLi** [comment]
Cool. Thanks for letting me know!

**Comment by hgaiser** [comment]
I assigned the label help wanted. We'd be happy to add pretrained (COCO/Pascal) networks to this repository if they are provided to us, but there is a risk that the architecture changes which causes those models to become obsolete. If that is the case, we likely won't update the pretrained models (except for ResNet50 on COCO).

**Comment by lvaleriu** [comment]
For training on coco what are the parameters? batch_size=1, flip_x  augmentation? (if that matters)

Yes, models might change a bit. It might be a good idea to use the official keras repository models (from applications), the ones from https://github.com/keras-team/keras-contrib or copy them directly in this repository (but we still need to link to the imagenet weights).

**Comment by hgaiser** [comment]
> For training on coco what are the parameters? batch_size=1, flip_x augmentation? (if that matters)

Yeah, only those.

**Comment by lvaleriu** [comment]
Started training on COCO (train2017+ val2017) using mobilenet224_1.0 + batch_size=1 + flip_x.
I'll keep updating this post with training results.

Epoch 1:
1/10000 [..............................] - ETA: 10:24:14 - loss: 5.3177 - regression_loss: 4.1452 - classification_loss: 1.1725
10000/10000 [==============================] - 1568s 157ms/step - loss: 4.4922 - regression_loss: 3.4150 - classification_loss: 1.0772

 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.001
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.004
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.001
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.002
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.002
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.027
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.057
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.062
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.006
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.033
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.118

**Comment by panda9095** [comment]
@lvaleriu  How is your mobilenet training process?  My training on COCO using densenet169 as backbone gives only a MAP of 0.028 at epoch 24.

**Comment by lvaleriu** [comment]
@panda9095  Very bad. So i'll start again using the FPN-CORRECTION.

**Comment by hgaiser** [comment]
Actually it got merged into `master`.

**Comment by smehida** [comment]
Could I use mobilennet initial weights from here?
https://github.com/experiencor/basic-yolo-keras
does it work?

**Comment by lvaleriu** [comment]
@panda9095 Started training mobilenet on coco again. I'll update the previous comment with the results after each epoch.

**Comment by lvaleriu** [comment]
@panda9095 It seems better now. 
@hgaiser Can you take a look at the learning progression? I've never trained resnet50 from scratch on coco till now and dont have a reference for the learning curve.

**Comment by jjiunlin** [comment]
Here are the results after training mobilenet224_1.0 for 140+ epoch(keras-retinanet0.2,batch_size=1)
Every epoch takes 60min on my single 1080ti.The GPU utilization is 90%+.
The red line is mobilenet224_1.0 and the orange line is res50_retinanet.It seems that the loss decrease very slow.
<img width="1026" alt="screen shot 2018-03-10 at 10 45 52 am" src="https://user-images.githubusercontent.com/33717622/37237596-60dfa48a-2450-11e8-9911-ca31cd05414e.png">
<img width="1038" alt="screen shot 2018-03-10 at 10 46 15 am" src="https://user-images.githubusercontent.com/33717622/37237605-74982628-2450-11e8-8d4e-d80042decc51.png">
<img width="1029" alt="screen shot 2018-03-10 at 10 46 33 am" src="https://user-images.githubusercontent.com/33717622/37237610-8015868a-2450-11e8-826a-d1c569162cfd.png">

**Comment by sujeet-gandhi** [comment]
@lvaleriu,  can u please explain why do we get 6 values of precision and recall?

**Comment by lvaleriu** [comment]
As defined in http://cocodataset.org/#detection-eval, here are the 12 metrics: 

![image](https://user-images.githubusercontent.com/927999/45032018-e58e9880-b050-11e8-810d-6e8639519f52.png)

**Comment by sujeet-gandhi** [comment]
@lvaleriu  Thanks.

**Comment by VCBE123** [comment]
Actually I need the more powerfull backbone support ,such as ResNeXt, or the SE-ResNeXt. Of course I tried by myself , but the performence dropped a litter while I excepted for higher. Maybe it's because that I used the customed  dataset which contains about 10K images. I will train on the COCO. If there is any idea for higher performence, I would be gratefull

**Comment by hgaiser** [comment]
PRs for those backbones would be very welcome.

Pretraining on COCO sounds like the right thing to do, it also gives you a better measure of how well the backbone works.

**Comment by TimoK93** [comment]
Hey all!
I just tried to train a net with mobilenet160_0.75 as backbone. I just added "--backbone mobilenet160_0.75" to the command provided in the README.md for training on csv datasets. It is throwing an error while creating mobilenet in site-package keras-applications. Did i forget an argument?

**Comment by hgaiser** [comment]
That's better suited for a separate issue (also, mention the error, it helps to find the cause).

**Comment by clw5180** [comment]
> Actually I need the more powerfull backbone support ,such as ResNeXt, or the SE-ResNeXt. Of course I tried by myself , but the performence dropped a litter while I excepted for higher. Maybe it's because that I used the customed dataset which contains about 10K images. I will train on the COCO. If there is any idea for higher performence, I would be gratefull

Do you take part in the competition of yuncong ?

**Comment by scstu** [comment]
@lvaleriu  could you please share with me the Pretrained models for mobilenet128_1.0 backbone ?

**Comment by liminghuiv** [comment]
For mobilenet, I saw keras-retinanet is used in vehicle detection:
https://github.com/yangliupku/retinanet_detection
Can someone merge it?

**Comment by hgaiser** [comment]
I'm closing this in favor of https://github.com/fizyr/keras-retinanet/issues/1161

**Comment by hasan-mh-aziz** [comment]
> Actually I need the more powerfull backbone support ,such as ResNeXt, or the SE-ResNeXt. Of course I tried by myself , but the performence dropped a litter while I excepted for higher. Maybe it's because that I used the customed dataset which contains about 10K images. I will train on the COCO. If there is any idea for higher performence, I would be gratefull

Where did you get the pretrained weights of ResNext? Which implementation of ResNext did you follow?

**Comment by Mansi2487** [comment]
> Actually I need the more powerfull backbone support ,such as ResNeXt, or the SE-ResNeXt. Of course I tried by myself , but the performence dropped a litter while I excepted for higher. Maybe it's because that I used the customed dataset which contains about 10K images. I will train on the COCO. If there is any idea for higher performence, I would be gratefull

Did you trained ResNeXt on COCO? If yes can you please provide me with the pretrained model.
